Initialize project with components

Kakalanp opened the pull request:
Adding the following components:
- Books
- Book
- Categories
- Navbar

index.js displays the navbar above everything else which can change the local path of the page on click, then calls App.js, which checks the path and renders the corresponding page.

Kakalanp committed:
Created react app

Kakalanp committed:
Added dependencies

Kakalanp committed:
Adding components and placeholders

Mihndim2020 commented on README.md:
```diff
@@ -1,2 +1,70 @@
+
+### `npm run build` fails to minify
+
+This section has moved here: [https://facebook.github.io/create-react-app/docs/troubleshooting#npm-run-build-fails-to-minify](https://facebook.github.io/create-react-app/docs/troubleshooting#npm-run-build-fails-to-minify)
```
- [x] Please kindly update your `readme file` to reflect the changes for this milestone, as this will make your `readme file` more informative and professional. Please checkout this [template ](https://github.com/microverseinc/readme-template)and adapt it to your project. The content of your `readme file` should be related to you and this project.

Kakalanp replied:
Forgot about the README! 🤣 

I'll update right away, thanks for the observation! 👍

Mihndim2020 commented on src/components/Books.js:
```diff
@@ -0,0 +1,43 @@
+          <Book key={book.id} title={book.title} author={book.author} category={book.category} />
+        ))}
+      </div>
+      <form>
+        <input type="text" />
+        <input type="text" />
+        <button type="submit" label="submit" />
+      </form>
```
- [x] As it is a [requirement](https://github.com/microverseinc/curriculum-react-redux/blob/main/bookstore/project_init_project_wth_comps_v2_0.md)(`Create a separate component for this form (with inputs for a title and an author`) for this project, and best practice, creating reusable components, please it will be nice to create a separate component for the form, then import and render it in this `Books.js` components.

Kakalanp replied:
Thanks for being so specific! I missed said requirement. 😅 

I'll be implementing the changes right away! 💯

Mihndim2020 reviewed (changes requested):
### Hi @Kakalanp , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages,
:heavy_check_mark:  Clean and organized code.

There are some issues that you still need to work on to get this project approved, but you are almost there!
![ALMOST-MINION](https://user-images.githubusercontent.com/65852621/133680327-431cb21d-ad7c-483a-84eb-21b733980a04.png)

# Status: Requires Changes ♻️
_Check the comments under the review._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Kakalanp committed:
Refactored Form, adding README, gh-pages

- Refactored Form to be a component called by Books.

- Adding a proper README.

- Adding gh-pages.

Kakalanp committed:
Update README.md

- It displays the correct GitHub pages now.

DeliceLydia reviewed (approved):
## Hi @Kakalanp, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Kakalanp commented:
Thanks for the approval! Merging right away! 🎉
